src/components/MapController/MapController.js

```diff
@@ -12,37 +13,68 @@ import styles from './MapController.css';
 
 var MapController = React.createClass({
 
-    propTypes: {
-        onSetArea: React.PropTypes.func.isRequired,
-    },
+  propTypes: {
+    variable: React.PropTypes.string,
+    meta: React.PropTypes.array,
+    onSetArea: React.PropTypes.func.isRequired,
+  },
 
   getInitialState: function () {
     return {
       styles: "boxfill/ferret",
-      timeofyear: "Annual",
-      time: "2000-01-01",
-      colorscalerange: "-50,11.0",
+      timeidx: 0,
       logscale: false
     }
   },
-  getDefaultProps: function() {
-    return {
-      variable: "tasmax",
-      dataset: "pr-tasmax-tasmin_day_BCSD-ANUSPLIN300-CanESM2_historical-rcp26_r1i1p1_19500101-21001231",
-    }
-  },
 
   updateSelection: function(param, selection) {
     var update = {}; update[param] = selection;
     this.setState(update);
   },
 
+  updateTime: function(timeidx) {
+
+    // Find selected dataset, then apply time transformation based on index
+    var selected = this.props.meta.filter(function(el){
+      return el.unique_id == this.state.dataset
+    }.bind(this))
+
+    // Assumes filter returns a single element which /should/ be true. FIXME.
+    this.setState({
+      timeidx: timeidx,
+      wmstime: selected[0].times[timeidx]})
+  },
+
+  updateDataset: function(dataset) {
+    // Updates dataset in state. Updates time value to match new dataset
+
+    var selected = this.props.meta.filter(function(el){
+      return el.unique_id == dataset
+    }.bind(this))[0]
+
+    this.setState({
+      dataset: dataset,
+      wmstime: selected.times[this.state.timeidx]
+    })
+  },
+
+  findUniqueId: function() {
+    if (this.props.meta.length > 0) {
+      return this.props.meta[0].unique_id;
+    }
+  },
+
   handleSetArea: function(wkt) {
-    // TODO: Do something more here?
-    // Really just here so it's not invisibly transferred from parent props, not directly needed
     this.props.onSetArea(wkt);
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    this.setState({
+      dataset: nextProps.meta[0].unique_id,
```

Comment:
Yes. It would be better to select the first dataset available, but we currently do not have a way to enforce default selection in the `Selector` component.